Lib/pathlib/__init__.py

```diff
@@ -209,7 +209,14 @@ def __hash__(self):
         try:
             return self._hash
         except AttributeError:
-            self._hash = hash(self._str_normcase)
+            hash_data = self._tail
+            if self._drv or self._root:
+                hash_data = [self._drv + self._root] + self._tail
+            elif self._tail and self.parser.splitdrive(self._tail[0])[0]:
+                hash_data = ['.'] + self._tail
```

Comment:
```suggestion
```

We don't need to add a leading dot for the hash value - it's a presentation detail for `__str__()` only.

Reply:
Actually, maybe we do need this so that `c:a` and `./c:a` hash unequal on Windows. TBH I regret adding support for NTFS file streams